src/Centreon/Infrastructure/MetaServiceConfiguration/Repository/MetaServiceConfigurationRepositoryRDB.php

```diff
@@ -0,0 +1,333 @@
+        $this->sqlRequestTranslator
+            ->getRequestParameters()
+            ->setConcordanceStrictMode(RequestParameters::CONCORDANCE_MODE_STRICT);
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function findAll(): array
+    {
+        return $this->findAllRequest(null);
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function findAllByContact(ContactInterface $contact): array
+    {
+        return $this->findAllRequest($contact->getId());
+    }
+
+    /**
+     * Find all meta services configurations filtered by contact id.
+     *
+     * @param int|null $contactId Contact id related to host categories
+     * @return MetaServiceConfiguration[]
+     * @throws AssertionFailedException
+     * @throws \InvalidArgumentException
+     */
+    private function findAllRequest(?int $contactId): array
+    {
+        $this->sqlRequestTranslator->setConcordanceArray([
+            'id' => 'ms.meta_id',
+            'name' => 'ms.meta_name',
+            'is_activated' => 'ms.meta_activate',
+        ]);
+        $this->sqlRequestTranslator->addNormalizer(
+            'is_activated',
+            new class implements NormalizerInterface
+            {
+                /**
+                 * @inheritDoc
+                 */
+                public function normalize($valueToNormalize)
+                {
+                    if (is_bool($valueToNormalize)) {
+                        return ($valueToNormalize === true) ? '1' : '0';
+                    }
+                    return $valueToNormalize;
+                }
+            }
+        );
+        if ($contactId === null) {
+            $request = $this->translateDbName(
+                "SELECT SQL_CALC_FOUND_ROWS
+                    ms.meta_id,
+                    ms.meta_name,
+                    ms.meta_display,
+                    ms.data_source_type,
+                    CASE
+                        WHEN ms.data_source_type = 0 THEN 'gauge'
+                        WHEN ms.data_source_type = 1 THEN 'counter'
+                        WHEN ms.data_source_type = 2 THEN 'derive'
+                        WHEN ms.data_source_type = 3 THEN 'absolute'
+                    END AS `data_source_type`,
+                    ms.meta_select_mode,
+                    ms.regexp_str,
+                    ms.metric,
+                    ms.warning,
+                    ms.critical,
+                    ms.meta_activate,
+                    ms.calcul_type,
+                    CASE
+                        WHEN ms.calcul_type = 'AVE' THEN 'average'
+                        WHEN ms.calcul_type = 'SOM' THEN 'sum'
+                        WHEN ms.calcul_type = 'MIN' THEN 'minimum'
+                        WHEN ms.calcul_type = 'MAX' THEN 'maximum'
+                    END AS `calculation_type`
+                FROM `:db`.meta_service ms"
+            );
+        } else {
+            $request = $this->translateDbName(
+                "SELECT SQL_CALC_FOUND_ROWS
```

Comment:
`SELECT` could be construct before `if` statement
Then, if contact id is set, add `JOIN`